### ISSUE TITLE
How to obtain the  memory table?

### ISSUE THREAD

**Issue by queenie88** [ryanaleksander/softened-similarity-learning#1]
Hi! If I use the other dataset, how to obtain the memory table. I can download your file, but I do not know how to construct the memory table.  Especially, we do not have the ground-truth labels, how to get the initial pseudo label for training.

**Comment by ryanaleksander** [comment]
To create a memory table you need 3 things: global vectors, part vectors, and camera IDs. First, you need to initialize a network with the argument pretrained=True. This will fill the backbone resnet50 architecture with pretrained weights. You can then pass images through the network. For each image, the network will return a parts vectors tensor, and a global vector tensor. You can then stack all global vector tensors and part vectors tensors, and use them together with a list of corresponding camera IDs to create a MemoryTable.

Pseudo labels are generated during the training process using the sample indices. So be careful not to enable shuffle for the DataLoader. If you want to use shuffle, a simple modification in the code should do. 

One more thing to notice is that in the class MemoryTable all the calculations are done on CPU, so all the tensors you pass in should also be loaded onto CPU. The reason for this is that MemoryTable requires a lot of...well..memory, and my GPU doesn't have a lot of GRAM. If you have enough GPU free real estate, you're welcome to change to code for the calculations to be done on GPU to boost training speed.

I hope this will help.
